src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
+
+  /**
+   * Given class which could correspond to generated lambda expressions etc infer appropriate description for it
+   *
+   * @param clazz
+   * @return description for task
+   */
+  String getDescription(Class<?> clazz);
```

Comment:
One thing I am wondering if we could change this interface to:
```
String getDescription(String clazz);
```
The reason is that we might want to use TaskDescriptor in the future for purposes other than this. For example, it would be cool use it to "fix up" stack traces of exceptions in ParSeq. For this purpose it would be more convenient to use just String to avoid loading Class by name.